### PR TITLE
Make pna.p4 dpdk specific and change methods like count and execute

### PR DESCRIPTION
**Pull request by kamleshbhalui**
as required by dpdk target

### CODE REVIEW

File: testdata/p4_16_samples/pna-action-selector-1.p4 [hunk 1]
```diff
@@ -15,7 +15,7 @@ limitations under the License.
 */
 
 #include <core.p4>
-#include "pna.p4"
+#include "dpdk/pna.p4"
```
Comment (psivanup) [inline testdata/p4_16_samples/pna-action-selector-1.p4:18]: I think, generic testcases shouldn't be modified with dpdk specific header.

**Review by psivanup — COMMENTED** [review]
Check the comment.

File: testdata/p4_16_samples/pna-action-selector-1.p4 (hunk 1 above)
Comment (kamleshbhalui) [reply testdata/p4_16_samples/pna-action-selector-1.p4]: done

**Comment by usha1830** [comment]
@kamleshbhalui We need to add new dpdk specific methods, not modify existing ones. Please check the description in https://jira.devtools.intel.com/browse/NVFMSA-5084.

File: testdata/p4_16_samples/psa-counter6.p4 [hunk 2]
```diff
@@ -50,7 +50,7 @@ control MyIC(
 
     DirectCounter<bit<12>>(PSA_CounterType_t.PACKETS) counter0;
     action execute() {
-        counter0.count();
```
Comment (psivanup) [inline testdata/p4_16_samples/psa-counter6.p4:53]: I guess this file is not for dpdk target.

File: p4include/dpdk/pna.p4 [hunk 3]
```diff
@@ -404,21 +406,25 @@ extern Meter<S> {
   // Use this method call to perform a color aware meter update (see
   // RFC 2698). The color of the packet before the method call was
   // made is specified by the color parameter.
+  PNA_MeterColor_t execute(in S index, in PNA_MeterColor_t color);
   PNA_MeterColor_t execute(in S index, in PNA_MeterColor_t color, in bit<32> pkt_len);
```
Comment (usha1830) [inline p4include/dpdk/pna.p4:410]: I think we should make this also as dpdk_execute

**Review by usha1830 — CHANGES_REQUESTED** [review]
Please see comments

**Review by usha1830 — COMMENTED** [review]
Please add the readme description as discussed and issue error for using execute

**Comment by usha1830** [comment]
@kamleshbhalui  Please also create a PR for pna and psa repositories for the same

File: backends/dpdk/README.md [hunk 4]
```diff
@@ -74,6 +74,13 @@ In P4 the second argument of the extract method is the number of bits.
 Compiler generates instructions which compute the size in bytes from the value in bits.
 If the value in bits is not a multiple of 8, the value is rounded down to the lower
 multiple of 8 bits.
+- Currently dpdk target does not support standard count and execute as defined in PSA and PNA, it requires packet length in methods count and execute which are part of Counter and Meter externs. Adding param ``` in bit<32> pkt_len``` to execute (part of Meter extern) leads to overload resolution failure due to ambiguous candidates, as p4c do overload resolution based on number of parameter and does not consider types. To workaround this we introduced new method in Meter extern `dpdk_execute` which has extra param.
```
Comment (usha1830) [inline backends/dpdk/README.md:77]: First sentence rephrased.
Currently dpdk target does not support standard count and execute **methods for Counter and Meter externs** as defined in PSA and PNA specifications.  It requires packet length as parameter in count and execute methods.

====== 
I think this detail is not necessary here as it is a user document. Instead it should be added as comment in the psa.p4 /pna.p4

> to execute (part of Meter extern) leads to overload resolution failure due to ambiguous candidates, as p4c do overload resolution based on number of parameter and does not consider types. To workaround this we introduced new method in Meter extern `dpdk_execute` which has extra param.

File: testdata/p4_16_samples/pna-example-tcp-connection-tracking-err-1.p4 [hunk 5]
```diff
@@ -15,7 +15,7 @@ limitations under the License.
 */
 
 #include <core.p4>
-#include "../pna.p4"
+#include "pna.p4"
```
Comment (usha1830) [inline testdata/p4_16_samples/pna-example-tcp-connection-tracking-err-1.p4:18]: This doesn't require change in path? dpdk/pna.p4?

File: testdata/p4_16_samples/psa-example-dpdk-byte-alignment_3.p4 [hunk 6]
```diff
@@ -69,7 +69,7 @@ control ingress(inout headers hdr, inout metadata_t user_meta, in psa_ingress_in
     PSA_MeterColor_t color_in = PSA_MeterColor_t.RED;
     action execute(bit<12> index) {
         hdr.ipv4.ihl = 5;
-        color_out = meter0.execute(index, color_in, hdr.ipv4.totalLen);
```
Comment (usha1830) [inline testdata/p4_16_samples/psa-example-dpdk-byte-alignment_3.p4:72]: Please sync with main, these tests have been updated recently

File: testdata/p4_16_samples/pna-example-tcp-connection-tracking-err-1.p4 (hunk 5 above)
Comment (psivanup) [reply testdata/p4_16_samples/pna-example-tcp-connection-tracking-err-1.p4:18]: @usha1830 , I think this is common test case and may not require dpdk/pna.p4.

**Comment by usha1830** [comment]
> @kamleshbhalui Please also create a PR for pna and psa repositories for the same

This is not required as we are not modifying original files. Sorry about the confusion

File: testdata/p4_16_samples/pna-example-tcp-connection-tracking-err-1.p4 (hunk 5 above)
Comment (usha1830) [reply testdata/p4_16_samples/pna-example-tcp-connection-tracking-err-1.p4:18]: Ok . Got it . Thanks!

**Review by usha1830 — APPROVED** [review]

**Review by psivanup — APPROVED** [review]
Looks fine to me

**Pull request closed without merging**